File: Problems/Leetcode/3001-4000/3201-3300/3248.py
from typing import List


class Solution:
    def finalPositionOfSnake(self, n: int, commands: List[str]) -> int:
        d = {"U": -n, "D": n, "L": -1, "R": 1}
        return sum(d[s[0]] for s in commands)

    def finalPositionOfSnake2(self, n: int, commands: List[str]) -> int:
        i = j = 0
        for s in commands:
            match s[0]:
                case "U":
                    i -= 1
                case "D":
                    i += 1
                case "L":
                    j -= 1
                case _:
                    j += 1  # 匹配其他任意值，相当于 switch 中的 default
        return i * n + j

File: Problems/Leetcode/3001-4000/3201-3300/3248.cpp
#include <bits/stdc++.h>

using namespace std;
using ll = long long;

static constexpr int MOD = 1e9 + 7;

class Solution {
public:
    int finalPositionOfSnake(int n, vector<string>& commands) {
        int x = 0, y = 0;
        for (auto& s : commands) {
            if (s == "RIGHT") {
                y++;
            } else if (s == "DOWN") {
                x++;
            } else if (s == "UP") {
                x--;
            } else {
                y--;
            }
        }
        return x * n + y;
    }
};

int main() {
    Solution sol;

    return 0;
}